[FIX] stock_split_picking: Fixed "split" when spliting move lines in stock.picking

maurochip opened the pull request:
Fixed a bug in the function `split` which ignored all the records in
`pack_operation_ids` except the first. Fixes issue #175.

maurochip committed:
[FIX] stock_split_picking: Fixed "split" when spliting move lines in stock.picking

Fixed a bug in the function `split` which ignored all the records in
`pack_operation_ids` except the first. Fixes issue #175.

rafaelbn commented:
Hi @maurochip , please check travis, it's failing

yvaucher commented:
This test is failing

```
2015-12-10 14:51:28,501 23059 DEBUG openerp_test openerp.tools.yaml_import: I check that the backorder has 40 units with state set to assigned not done
2015-12-10 14:51:28,508 23059 ERROR openerp_test openerp.tools.yaml_import: AssertionError in Python code  (line 69): Backorder should be assigned, not u'confirmed'.
```

yvaucher commented:
@maurochip the commits is not authored correctly, you might need to do `git commit --amend --author="Name <email>"`

rafaelbn commented:
cc @JMODaERP

JMODaERP commented:
I tested in runbot :+1: 

Thanks..

yvaucher commented:
@maurochip Please do a `git rebase -i 87e860025f66c30c5fcf58fbb7a1b82980d04771~1` and merge your commits using "fixup"

To update this PR you can do a `git push -f`

maurochip commented:
Sorry for the errors, please check all is correct, thanks.

yvaucher commented:
I tested a bit on runbot I have strange issues.

I add 300 M-Wir to my inventory
I create a picking with 50 M-Wir and 100 Ice creams
I mark as todo
I check availibity
I split 25 M-Wir and 20 Ice creams
My picking gets a remaining of 25 M-Wir and a new picking has been created with 25 M-Wir and 100 Ice creams.

It ignores my split for second line.

maurochip commented:
The error that you describes is the error that happens without my change because it was deleting the pack operations. With my change, my tests work well and not ignores the second line in the split.

yvaucher commented:
@maurochip Maybe something failed on the runbot, I'll have an other try later this week locally.
Could you add a test in yaml or python unit test, testing this case ?.

maurochip committed:
[ADD] Added unit test to check stock split with two move lines

maurochip commented:
Can anyone take a look at this, please? I cannot fix some of the Travis tests, but they are unrelated to this fix.

yvaucher commented:
@maurochip 
The error is related to this PR

It fails on stock_split_picking/test/test_assigned_picking_split.yml

```
I check that the backorder has 40 units with state set to assigned not done
AssertionError in Python code  (line 69): Backorder should be assigned, not u'confirmed'.
```

maurochip commented:
I'm executing the same tests in my local odoo installation and they do not show any error, can you manually execute them please? Thank you.

P4R committed:
[8.0][FIX] product_unique_serial:Change on check quantity unicity

Check quantity unicity was calculated using quanty on hand, now it check qty using the sum of quant's lot.
Moreover I remove the parameter lot_id in mothods: check_after_action_done and check_unicity_qty_available.

P4R committed:
[MOD] Back to old condition.

P4R committed:
[REM] Removed pdb

P4R committed:
[FIX] Solved some bugs to pass somo test.
[REM] Removed lot_id parameter in check_before_action_done because it's unnecessary.

P4R committed:
flake8

P4R committed:
[MOD] Variable convention

P4R committed:
[REM] Removed wrong test.

hbrunn commented:
the only commit you actually need is https://github.com/OCA/stock-logistics-workflow/pull/185/commits/3d6c118156bd29ddbda7fbafba39dad4858a8355

laetitia-gangloff commented:
And this one too : 9820542b3d628c1f42fee24f33278e83e251be64 (for the test).

hbrunn commented:
right, the test. @laetitia-gangloff given @maurochip didn't reply for months and there seem to be issues with the CLA anyways, will you do the honors to propose a clean PR for this, then we can close this one?

laetitia-gangloff commented:
Yes, I can !
And I already have a branch  with these 2 commits ;)

hbrunn commented:
well, perfect, thanks! Closing this one
